src/openrct2/ride/TrackDesign.cpp

```diff
@@ -247,7 +247,6 @@ rct_string_id TrackDesign::CreateTrackDesignTrack(const Ride& ride)
         }
         trackElement.x = newCoords->x;
         trackElement.y = newCoords->y;
-        z = newCoords->z;
```

Comment:
`z` is used on line 290 - removing the statement changes the value of it.

Reply:
`z` gets assigned again before that on line 263. The value here is completely discarded.